libraries/AP_Param/AP_Param.h

```diff
@@ -542,11 +542,16 @@ class AP_ParamT : public AP_Param
             set(v);
         }
     }
-    
+
     /// Value setter - set value, tell GCS
     ///
     void set_and_notify(const T &v) {
+// We do want to compare each value, even floats, since it being the same here
+// is the result of previously setting it.
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wfloat-equal"
         if (v != _value) {
+#pragma GCC diagnostic pop
```

Comment:
I thought this template version was also used for Vector. Now I see it's a different one... trying to use is_equal() here is creating a more handful of commits due to include order, missing definitions... I may just give up... that's probably why below it's just using float epsilon rather than trying to include AP_Math.h :(

> And I'm surprised there's no issue with the set_and_save_ifchanged below.

Probably because it's never called for floats and thus not instantiated, not generating warnings